File: My Leet Code Python Solutions/Solution1491.py
#1491. Average Salary Excluding the Minimum and Maximum Salary

class Solution:
    def average(self, salary: List[int]) -> float:
        max = salary[0]
        min = salary[0]
        sum = 0
        counter = 0
        for i in range(1, len(salary)):
            if salary[i] > max:
                max = salary[i]
            if salary[i] < min:
                min = salary[i]
        for i in range(0, len(salary)):
            if max == salary[i] or min == salary[i]:
                continue
            else:
                sum += salary[i]
                counter += 1
        return sum/counter

File: My Leet Code Java Solutions/Solution1491.java
//1491. Average Salary Excluding the Minimum and Maximum Salary

class Solution1491 {
    public double average(int[] salary) {
        int min = salary[0];
        int max = salary[0];
        int sum = salary[0];
        for (int i = 1; i < salary.length; i++) {
            if (salary[i] < min) { // salary[i]=100,min=10
                min = salary[i];
            } else if (salary[i] > max) { // salary[i]=100,max=100
                max = salary[i];
            }
            sum += salary[i];
        }
        return (double) (sum - min - max) / (salary.length - 2);
    }
}